File: Cpp/2085_count_common_words_with_one_occurrence.cpp
// https://leetcode.cn/problems/count-common-words-with-one-occurrence/

class Solution {
public:
    int countWords(vector<string>& words1, vector<string>& words2) {
        unordered_map<string, int> cnt1, cnt2;
        for (auto &word: words1) {
            cnt1[word] += 1;
        }
        for (auto &word: words2) {
            cnt2[word] += 1;
        }
        int res = 0;
        for (auto it = cnt1.begin(); it != cnt1.end(); ++it) {
            if (it->second == 1 && cnt2[it->first] == 1) {
                res += 1;
            }
        }
        return res;
    }
};

File: Python3/2085_count_common_words_with_one_occurrence.py
# https://leetcode.cn/problems/count-common-words-with-one-occurrence/

class Solution:
    def countWords(self, words1, words2):
        """
        Inputs:
            words1, words2: list[str]
        Outputs:
            res: int
        """
        cnt1, cnt2 = dict(), dict()
        for word in words1:
            cnt1[word] = cnt1.get(word,0) + 1
        for word in words2:
            cnt2[word] = cnt2.get(word,0) + 1
        
        res = 0
        for k,v in cnt1.items():
            if v == 1 and cnt2.get(k,0) == 1:
                res += 1
        return res